src/releaseNotesProvider.ts

```diff
@@ -0,0 +1,180 @@
+import { env, ExtensionContext } from "vscode";
+import * as vscode from "vscode";
+import * as semver from "semver";
+import { Remarkable } from "remarkable";
+import http from "https";
+
+const versionKey = "metals-server-version";
+
+// prettier-ignore
+const releaseNotesMarkdownUrl: Record<string, string> = {
+  "0.11.5": "https://raw.githubusercontent.com/scalameta/metals/main/website/blog/2022-04-28-aluminium.md",
+  "0.11.6": "https://raw.githubusercontent.com/scalameta/metals/main/website/blog/2022-06-03-aluminium.md",
+};
+
+/**
+ * Show release notes if possible, swallow errors since its not a crucial feature.
+ * Treats snapshot versions like 0.11.6+67-926ec9a3-SNAPSHOT as a 0.11.6.
+ */
+export async function showReleaseNotesIfNeeded(
+  context: ExtensionContext,
+  serverVersion: string,
+  outputChannel: vscode.OutputChannel
+) {
+  try {
+    await showReleaseNotes(context, serverVersion);
+  } catch (error) {
+    outputChannel.appendLine(
+      `Couldn't show release notes for Metals ${serverVersion}`
+    );
+    outputChannel.appendLine(`${error}`);
+  }
+}
+
+async function showReleaseNotes(
+  context: ExtensionContext,
+  currentVersion: string
+): Promise<void> {
+  const state = context.globalState;
+
+  const version = getVersion();
+  const releaseNotesUrl = version
+    ? releaseNotesMarkdownUrl[version]
+    : undefined;
```

Comment:
There's a tricky part here, locally published snapshot will be trimmed to 'major.minor.patch' but there won't be released notes for such version -> `state.update(versionKey, newVersion);` won't trigger until there will be.

Reply:
I should add comment with this